040-SDK/160-image-transformations.mdx

```diff
@@ -0,0 +1,881 @@
+---
+title: Image transformations
+navTitle: Image transformations
+keywords: ['images', 'image resize', 'file attachments']
+description: Transform the shape, quality, size and other attributes for any image in Xata
+slug: sdk/image-transformations
+published: true
+---
+
+For the very common case of storing images, Xata offers a set of transformations that can be applied in real-time while retrieving images through [access URLs](/docs/concepts/file-attachments#file-access-urls). Transformations are specified in the URL and their result is cached by the CDN. Both public and signed URLs support transformations. In the URL the transformation specification is a comma separated list of individual transformations following the `/transform/` segment.
+
+Original image URL:
+
+https://us-east-1.storage.xata.sh/4u1fh2o6p10blbutjnphcste94
+
+Image URL with transformations:
+
+https://us-east-1.storage.xata.sh/transform/rotate=180,height=100,format=webp/4u1fh2o6p10blbutjnphcste94
+
+<TransformImageExample
+  url="https://us-east-1.storage.xata.sh/4u1fh2o6p10blbutjnphcste94"
+  options={{ rotate: 180, height: 100 }}
+/>
+
+The below example shows a method to transform an image directly off the file object in a record.
+
+<TabbedCode tabs={['TypeScript', 'Python', 'URL']}>
+
+```ts
+// Get a specific record from the "gallery" database
+const record = xata.db.gallery.read('record_id');
```

Comment:
I also think this may need `await`

Reply:
Well look at Mr. Engineer over here! Nice spot.